@here/harp-map-controls/lib/MapControlsUI.ts

```diff
@@ -29,15 +29,21 @@ export class MapControlsUI {
     /**
      * The DOM node containing the UI.
      */
-    domElement: HTMLDivElement;
+    readonly domElement = document.createElement("div");
+
+    /**
+     * Displays zoom level if [[MapControlsUIOptions.zoomLevel]] is defined.
+     */
+    private zoomLevelElement: HTMLDivElement | HTMLInputElement | null = null;
 
     /**
      * Constructor of the UI.
      *
      * @param controls Controls referencing a [[MapView]].
      */
     constructor(readonly controls: MapControls, options: MapControlsUIOptions = {}) {
-        this.domElement = document.createElement("div");
+        this.onMapViewRenderEvent = this.onMapViewRenderEvent.bind(this);
```

Comment:
Style - Please define listeners that have to be bound to this as "fat arrow member functions" i.e:
```
private onMapViewRenderEvent = (...) => {
}
```